pkg/kudoctl/util/repo/repo.go

```diff
@@ -1,13 +1,92 @@
 package repo
 
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/afero"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// Version is the repo / packaging version
+	Version         = "v1"
+	defaultRepoName = "community"
+)
+
 // RepositoryConfiguration represents a collection of parameters for operator repository.
 type RepositoryConfiguration struct {
-	URL string `json:"url"`
+	URL  string `json:"url"`
+	Name string `json:"name"`
+}
+
+// Repositories represents the repositories.yaml file usually in the $KUDO_HOME
+type Repositories struct {
+	RepoVersion  string                     `json:"repoVersion"`
+	Context      string                     `json:"context"`
+	Repositories []*RepositoryConfiguration `json:"repositories"`
 }
 
 // Default initialized repository.
 var Default = &RepositoryConfiguration{
-	URL: "https://kudo-repository.storage.googleapis.com",
+	Name: defaultRepoName,
+	URL:  "https://kudo-repository.storage.googleapis.com",
+}
+
+// NewRepoFile creates a new repo with only defaults populated
+func NewRepoFile() *Repositories {
+	return &Repositories{
+		RepoVersion:  Version,
+		Context:      defaultRepoName,
+		Repositories: []*RepositoryConfiguration{Default},
+	}
+}
+
+// GetRepo returns a RepoName Config for a name or nil
+func (r Repositories) GetRepo(name string) *RepositoryConfiguration {
+	for _, repo := range r.Repositories {
+		if repo.Name == name {
+			return repo
+		}
+	}
+	return nil
+}
+
+// CurrentRepo provides the repo config for the current context
+func (r Repositories) CurrentRepo() *RepositoryConfiguration {
+	return r.GetRepo(r.Context)
+}
+
+// LoadRepositories reads the Repositories file
+func LoadRepositories(fs afero.Fs, path string) (*Repositories, error) {
+	b, err := afero.ReadFile(fs, path)
+	if err != nil {
+		if os.IsNotExist(err) {
```

Comment:
What about using the prior Stat/Exists func before attempting to read, then give the contextual error? Cleans up the nested if block.